### PR TITLE
Add cancel and complete

### PR DESCRIPTION
**Pull request by adamradocz**
Add CompleteAsync and CancelPending to the Protocols.

### CODE REVIEW

**Comment by adamradocz** [comment]
@davidfowl I just realised that, this PR also contains the modifications of #124.  Do you want me separate it from that one?

Other:
Related to #127, I'd like to make a PR that updates the client sample app to show how to disconnect from the server. Is that OK with you? Just asking, because this repo seems abandoned to me.

File: src/Bedrock.Framework.Experimental/Bedrock.Framework.Experimental.csproj [hunk 1]
```diff
@@ -18,6 +18,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Http.Connections.Client" Version="3.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Connections.Client" Version="6.0.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Update="Nerdbank.GitVersioning" Version="3.4.244" />
```
Comment (davidfowl) [inline src/Bedrock.Framework.Experimental/Bedrock.Framework.Experimental.csproj:25]: Why is this here?

File: src/Bedrock.Framework.Experimental/Protocols/Http1RequestMessageReader.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ public bool TryParseMessage(in ReadOnlySequence<byte> input, ref SequencePositio
                     goto case State.Headers;
 
                 case State.Headers:
-                    while (sequenceReader.TryReadTo(out var headerLine, NewLine))
+                    while (sequenceReader.TryReadTo(out ReadOnlySequence<byte> headerLine, NewLine))
```
Comment (davidfowl) [inline src/Bedrock.Framework.Experimental/Protocols/Http1RequestMessageReader.cs:57]: Undo

File: src/Bedrock.Framework.Experimental/Protocols/Http1ResponseMessageReader.cs [hunk 3]
```diff
@@ -59,7 +59,7 @@ public bool TryParseMessage(in ReadOnlySequence<byte> input, ref SequencePositio
                     goto case State.Headers;
 
                 case State.Headers:
-                    while (sequenceReader.TryReadTo(out var headerLine, NewLine))
+                    while (sequenceReader.TryReadTo(out ReadOnlySequence<byte> headerLine, NewLine))
```
Comment (davidfowl) [inline src/Bedrock.Framework.Experimental/Protocols/Http1ResponseMessageReader.cs:62]: Undo

File: src/Bedrock.Framework/Bedrock.Framework.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (davidfowl) [inline src/Bedrock.Framework/Bedrock.Framework.csproj:4]: This should support both tfms

File: src/Bedrock.Framework/Bedrock.Framework.csproj [hunk 5]
```diff
@@ -12,6 +12,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.IO.Pipelines" Version="4.7.2" />
+    <PackageReference Include="System.IO.Pipelines" Version="6.0.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Update="Nerdbank.GitVersioning" Version="3.4.244" />
```
Comment (davidfowl) [inline src/Bedrock.Framework/Bedrock.Framework.csproj:19]: This shouldn't be here.

File: src/Bedrock.Framework.Experimental/Bedrock.Framework.Experimental.csproj (hunk 1 above)
Comment (adamradocz) [reply src/Bedrock.Framework.Experimental/Bedrock.Framework.Experimental.csproj:25]: It is in the master too. I just updated it.

File: src/Bedrock.Framework/Bedrock.Framework.csproj (hunk 5 above)
Comment (adamradocz) [reply src/Bedrock.Framework/Bedrock.Framework.csproj:19]: It is in the master too. I just updated it.

File: src/Bedrock.Framework.Experimental/Protocols/Http1ResponseMessageReader.cs (hunk 3 above)
Comment (adamradocz) [reply src/Bedrock.Framework.Experimental/Protocols/Http1ResponseMessageReader.cs:62]: On .NET 6 you get the following error when you use the `var`:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0121	The call is ambiguous between the following methods or properties: 'SequenceReader<T>.TryReadTo(out ReadOnlySequence<T>, ReadOnlySpan<T>, bool)' and 'SequenceReader<T>.TryReadTo(out ReadOnlySpan<T>, ReadOnlySpan<T>, bool)'	Bedrock.Framework.Experimental	c:\Users\radocz.adam\source\repos\Bedrock\src\Bedrock.Framework.Experimental\Protocols\Http1RequestMessageReader.cs	57	Active
```

File: src/Bedrock.Framework.Experimental/Protocols/Http1RequestMessageReader.cs (hunk 2 above)
Comment (adamradocz) [reply src/Bedrock.Framework.Experimental/Protocols/Http1RequestMessageReader.cs:57]: On .NET 6 you get the following error when you use the `var`:
```
Error	CS0121	The call is ambiguous between the following methods or properties: 
'SequenceReader<T>.TryReadTo(out ReadOnlySequence<T>, ReadOnlySpan<T>, bool)' and
'SequenceReader<T>.TryReadTo(out ReadOnlySpan<T>, ReadOnlySpan<T>, bool)'
```

File: src/Bedrock.Framework/Bedrock.Framework.csproj (hunk 4 above)
Comment (adamradocz) [reply src/Bedrock.Framework/Bedrock.Framework.csproj]: Fixed it.
The only drawback is you can update the `Microsoft.AspNetCore.Http.Connections.Client` package to v5.0.12 and not v6.0.0.
